sdk/src/azure/core/az_mqtt5_connection_hfsm.c

```diff
@@ -6,19 +6,36 @@
 #include <azure/core/az_platform.h>
 #include <azure/core/az_result.h>
 #include <azure/core/internal/az_log_internal.h>
+#include <stdlib.h>
 
 #include <azure/core/_az_cfg.h>
 
 static az_result root(az_event_policy* me, az_event event);
 
 static az_result idle(az_event_policy* me, az_event event);
 static az_result started(az_event_policy* me, az_event event);
-// static az_result faulted(az_event_policy* me, az_event event);
+static az_result faulted(az_event_policy* me, az_event event);
 
 static az_result connecting(az_event_policy* me, az_event event);
 static az_result connected(az_event_policy* me, az_event event);
 static az_result disconnecting(az_event_policy* me, az_event event);
-// static az_result reconnect_timeout(az_event_policy* me, az_event event);
+static az_result reconnect_timeout(az_event_policy* me, az_event event);
+
+static void _connection_timeout_callback(void* context)
+{
+  az_mqtt5_connection* this_policy = (az_mqtt5_connection*)context;
+
+  if (_az_hfsm_send_event(
+          (_az_hfsm*)this_policy,
+          (az_event){
+              .type = AZ_HFSM_EVENT_TIMEOUT,
+              .data = NULL,
```

Comment:
On a timeout event, I think the data is expected to be the timer

Reply:
Removed callback, using the pipeline based one now